File: 2020/25/25.py
def generator(input):
    return [int(item) for item in input.splitlines()]

def part_1(input):
    p, q = input
    k = 1
    while True:
        key = pow(7, k, 20201227)
        if key == p: return pow(q, k, 20201227)
        if key == q: return pow(p, k, 20201227)
        k += 1

def part_2(input): return "MERRY X-MAS"

File: 2015/25/25.rs
aoc::main!();

fn preprocessing(input: &str) -> (u64, u64) {
    input.split([',',' ','.']).filter_map(|n| n.parse().ok()).collect_tuple().unwrap()
}

fn part_1(coord :(u64, u64)) -> u64 {
    20151125 * mod_exp(252533, (coord.0 + coord.1)*(coord.0 + coord.1 - 1) / 2 - coord.0, 33554393) % 33554393
}

fn part_2(_coord : (u64, u64)) -> String {
    "Merry XMAS".to_string()
}